programs/mango-v4/src/instructions/token_withdraw.rs

```diff
@@ -59,8 +59,13 @@ pub fn token_withdraw(ctx: Context<TokenWithdraw>, amount: u64, allow_borrow: bo
     require_msg!(amount > 0, "withdraw amount must be positive");
 
     let group = ctx.accounts.group.load()?;
+    let mut bank = ctx.accounts.bank.load_mut()?;
     let token_index = ctx.accounts.bank.load()?.token_index;
 
+    if bank.is_reduce_only() {
+        require!(!allow_borrow, MangoError::SomeError);
+    }
```

Comment:
we should allow withdraws that don't trigger borrows, even if allow_borrow is true

Reply:
done